pkg/report/report.go

```diff
@@ -425,9 +442,40 @@ func Generate(ctx context.Context, path string, mrs yara.MatchRules, c malconten
 				syscalls = append(syscalls, sy...)
 			case "cap":
 				caps = append(caps, v)
+			// Reduce the severity by one level if the path contains `v`
+			case "decrease_if":
+				if strings.Contains(path, v) {
+					newSev := b.RiskScore - 1
+					b.RiskScore = newSev
+					b.RiskLevel = RiskLevels[newSev]
+				}
+			// Ignore this rule if `drop_if` is populated and the path contains `v`
+			case "drop_if":
+				if v != "" && strings.Contains(path, v) {
+					skipMatch = true
+				}
+			// If we find a rule to override, set that rule's risk score and level
+			case overrideRule:
+				var overrideSev int
+				if _, ok := Levels[v]; ok {
+					overrideSev = Levels[v]
+				}
+				// Find the behavior for the overridden rule
+				for _, b := range fr.Behaviors {
```

Comment:
This does work:
```
go run . analyze out/samples-75ee4b4f3e5c29484811e77ffa9d5d02eb75218f/javascript/clean/connection.js
out/samples-75ee4b4f3e5c29484811e77ffa9d5d02eb75218f/javascript/clean/connection.js [⚠️ MEDIUM]
---------------------------------------------------------------------------------------------------------------
RISK  KEY                         DESCRIPTION                      EVIDENCE                                    
---------------------------------------------------------------------------------------------------------------
LOW   encoding/base64             Supports base64 encoded strings  base64                                      
LOW   fd/write                    writes to a file handle          outgoingMessageStream.write(message)        
LOW   net/dns                     Uses DNS (Domain Name Service)   require("dns")                              
LOW   ref/site/url                contains embedded HTTPS URLs     https://github.com/pekim/tedious/issues/24  
LOW   ref/words/password          references a 'password'          UsernamePasswordCredential                  
                                                                   directory-password                          
MED   data/embedded/base64/terms  Contains base64 content          FkZHJlc3::$address                          
                                                                   RpcmVjdG9ye::$directory                     
                                                                   YWRkcmVzc::$address                         
                                                                   ZGlyZWN0b3J5::$directory                    
                                                                   aHRtb::$html                                
                                                                   hZGRyZXNz::$address                         
                                                                   kaXJlY3Rvcn::$directory                     
MED   data/embedded/base64/url    Contains base64 url              aHR0cDovL::$http                            
                                                                   h0dHA6Ly::$http                             
                                                                   h0dHBzOi8v::$https                          
                                                                   odHRwOi8v::$http                            
                                                                   odHRwczovL::$https                          
MED   evasion/hex                 converts hex data to ASCII       toString('hex');                            
CRIT  net/socket/send             send a message to a socket       send                                        
                                                                   socket                                      
---------------------------------------------------------------------------------------------------------------
```

However, and I may be overthinking this (or it may not happen?), there could be an order of operations issue that evaluates the override rule before the rule it overrides which would leave said rule unmodified.